test/boost/aws_error_injection_test.cc

```diff
@@ -0,0 +1,240 @@
+                                     <Bucket>Example-Bucket</Bucket>
+                                     <Key>Example-Object</Key>
+                                     <ETag>"3858f62230ac3c915f300c664312c11f-9"</ETag>
+                                </CompleteMultipartUploadResult>)";
+                    case failure_policy::RETRYABLE_FAILURE:
+                        // should succeed on retry
+                        _test_server.test_failure_policy = failure_policy::SUCCESS;
+                        assert(aws::aws_error_map.at("InternalError").is_retryable());
+                        return R"(<?xml version="1.0" encoding="UTF-8"?>
+
+                                 <Error>
+                                  <Code>InternalError</Code>
+                                  <Message>We encountered an internal error. Please try again.</Message>
+                                  <RequestId>656c76696e6727732072657175657374</RequestId>
+                                  <HostId>Uuag1LuByRx9e6j5Onimru9pO4ZVKnJ2Qz7/C1NPcfTWAtRPfTaOFg==</HostId>
+                                 </Error>)";
+                    case failure_policy::NONRETRYABLE_FAILURE:
+                        assert(!aws::aws_error_map.at("InvalidAction").is_retryable());
+                        return R"(<?xml version="1.0" encoding="UTF-8"?>
+
+                                 <Error>
+                                  <Code>InvalidAction</Code>
+                                  <Message>Something went terribly wrong</Message>
+                                  <RequestId>656c76696e6727732072657175657374</RequestId>
+                                  <HostId>Uuag1LuByRx9e6j5Onimru9pO4ZVKnJ2Qz7/C1NPcfTWAtRPfTaOFg==</HostId>
+                                 </Error>)";
+                    }
+                }
+            }
+            return response;
+        }
+        server& _test_server;
+    };
+    explicit server(failure_policy _test_failure_policy) : test_failure_policy(_test_failure_policy) {
+        handler = std::make_unique<dummy_aws_request_handler>(*this);
+    }
+    future<> start() {
+        co_await seastar::async([this] {
+            uint16_t port = 2012;
+            net::inet_address addr("127.0.0.1");
+            test_server.start("test").get();
+
+            test_server.server()
+                .invoke_on_all([this](httpd::http_server& server) {
+                    server._routes.add_default_handler(handler.get());
+                    return make_ready_future<>();
+                })
+                .get();
+
+            test_server.listen(socket_address{addr, port}).handle_exception([](auto ep) { return make_exception_future<>(ep); }).get();
+        });
+    }
+    future<> stop() {
+        test_server.server().stop().get();
+        return test_server.stop();
+    }
+
+private:
+    std::unique_ptr<httpd::handler_base> handler;
+    httpd::http_server_control test_server;
```

Comment:
better off renaming `test_server` to `_http_server`. as this server is for testing. then one would wonder what a `test_server` in a server for testing is for?

Reply:
done